src/sentry/tagstore/snuba/backend.py

```diff
@@ -474,13 +488,13 @@ def get_tag_value_paginator(self, project_id, environment_id, key, query=None,
         if query:
             conditions.append(['tags_value', 'LIKE', u'%{}%'.format(query)])
 
-        start, end = self.get_time_range()
         filters = {
-            'project_id': [project_id],
+            'project_id': projects,
             'tags_key': [key],
```

Comment:
Dont have to change this here, but I think we can make this query potentially a lot more efficient if, instead of `WHERE tags_key = 'key'` we use `WHERE tags[key] != ''`, as the former will always do an arrayjoin on the entire set of tags, and the latter can use the fast path if the tag is promoted. cc @bretthoerner who wrote the original snuba version of `get_tag_value_paginator` in case I'm completely missing something though.

Reply:
Perhaps, though, if we did change it now, it would end up helping with your other problem. If we did something more like [this](https://github.com/getsentry/sentry/blob/master/src/sentry/tagstore/snuba/backend.py#L178-L186) then we could more easily make it work for the other columns that are not technically tags (eg. `http_referrer`) by simply not wrapping them in `tags[]` and all the other code would remain the same.

Reply:
That sounds right to me, I think it was just an optimization oversight, so long as it effectively does the same thing it sounds good.